backend/services/calculateMatchScore.js

```diff
@@ -24,29 +39,66 @@ function calculateMatchScore(currentUser, streamerToCompare) {
   const ageDifference = Math.abs(currentUserAge - streamerAge);
 
   if (ageDifference <= VERY_CLOSE_AGE_THRESHOLD) {
-    score += 4;
+    return 4;
   } else if (ageDifference <= CLOSE_AGE_THRESHOLD) {
-    score += 2;
+    return 2;
   } else if (ageDifference <= SOMEWHAT_CLOSE_AGE_THRESHOLD) {
-    score++;
+    return 1;
   }
+  return 0.5;
+}
 
+function calcAudienceScore(currentUser, streamerToCompare) {
   //check targetAudience. This one is a BIG boost. If user and streamer we are comparing them with stream to same audience give them 6 points
   if (currentUser.targetAudience === streamerToCompare.targetAudience) {
-    score += 6;
+    return 5;
+  } else {
+    return 1; //small base score if audience isnt the saem but at least weights will still affect it.
   }
+}
 
+function calcTagScore(currentUser, streamerToCompare) {
   //TODO: CASE INSENSITIVITY if user has tag "valorant" vs "Valorant" should still match
   //Now we need to check tags. We can add points depending on how many tags are shared between the user and the streamer we are checking
   //filter through current users tags and return tags that are in the streamerTocompare tags
-  const sharedTags = currentUser.tags.filter( (tag) => {
+  const sharedTags = currentUser.tags.filter((tag) => {
     return streamerToCompare.tags.includes(tag);
-  })
+  });
+
+  //for each tag that is in common add 2 points to the score. Hence the sharedTags * 2
+  return Math.max(1, sharedTags.length * 2); //return min 1 point so it isnt 0 if no shared tags
+}
 
-  //for each tag that is in common add 3 points to the score. Hence the sharedTags * 3
-  score += sharedTags.length * 3;
+function calcGameScore(currentUser, streamerToCompare) {
+  // Handle missing game data not all twitch users might have it most do though
+  if (!currentUser.twitch_game_name || !streamerToCompare.twitch_game_name) {
+    return 1; // Can't compare, return base score
+  }
+
+  if (currentUser.twitch_game_name === streamerToCompare.twitch_game_name) {
+    return 3; // Same game
+  } else {
+    return 1; // Different game
```

Comment:
Why do we want to return 1 if there are no match here? Same with other no-matching default values